airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/MySqlQueryUtils.java

```diff
@@ -0,0 +1,73 @@
+package io.airbyte.integrations.source.mysql;
+
+import static io.airbyte.integrations.source.relationaldb.RelationalDbQueryUtils.getFullyQualifiedTableNameWithQuoting;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.base.Preconditions;
+import io.airbyte.db.jdbc.JdbcDatabase;
+import io.airbyte.db.jdbc.JdbcUtils;
+import io.airbyte.protocol.models.AirbyteStreamNameNamespacePair;
+import io.airbyte.protocol.models.v0.ConfiguredAirbyteStream;
+import java.sql.SQLException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class MySqlQueryUtils {
+  private static final Logger LOGGER = LoggerFactory.getLogger(MySqlQueryUtils.class);
+  public record TableSizeInfo(Long tableSize, Long avgRowLength) { }
+
+  public static final String TABLE_ESTIMATE_QUERY =
```

Comment:
We are sure that this query is not going to cause any trouble (getting stuck) with huge tables right?

Reply:
This should be fast since it's querying information_schema.tables. 
The drawback here is that it is just an estimate, which should be ok for our use case